Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/FixedPlaneConstraint.inl

```diff
@@ -241,50 +289,14 @@ void FixedPlaneConstraint<DataTypes>::draw(const VisualParams* vparams)
     vparams->drawTool()->drawPoints(points, 10, sofa::type::RGBAColor{1,1.0,0.5,1});
 }
 
-/// This function are there to provide kind of type translation to the vector one so we can
-/// implement the algorithm as is the different objects where of similar type.
-/// this solution is not really satisfactory but for the moment it does the job.
-/// A better solution would that all the used types are following the same iterface which
-/// requires to touch core sofa classes.
-sofa::type::Vec3d& getVec(sofa::defaulttype::Rigid3dTypes::Deriv& i){ return i.getVCenter(); }
-sofa::type::Vec3d& getVec(sofa::defaulttype::Rigid3dTypes::Coord& i){ return i.getCenter(); }
-const sofa::type::Vec3d& getVec(const sofa::defaulttype::Rigid3dTypes::Coord& i){ return i.getCenter(); }
-sofa::type::Vec3d& getVec(sofa::defaulttype::Vec3dTypes::Deriv& i){ return i; }
-const sofa::type::Vec3d& getVec(const sofa::defaulttype::Vec3dTypes::Deriv& i){ return i; }
-sofa::type::Vec6d& getVec(sofa::defaulttype::Vec6dTypes::Deriv& i){ return i; }
-const sofa::type::Vec6d& getVec(const sofa::defaulttype::Vec6dTypes::Deriv& i){ return i; }
-
-sofa::type::Vec3f& getVec(sofa::defaulttype::Rigid3fTypes::Deriv& i){ return i.getVCenter(); }
-sofa::type::Vec3f& getVec(sofa::defaulttype::Rigid3fTypes::Coord& i){ return i.getCenter(); }
-const sofa::type::Vec3f& getVec(const sofa::defaulttype::Rigid3fTypes::Coord& i){ return i.getCenter(); }
-sofa::type::Vec3f& getVec(sofa::defaulttype::Vec3fTypes::Deriv& i){ return i; }
-const sofa::type::Vec3f& getVec(const sofa::defaulttype::Vec3fTypes::Deriv& i){ return i; }
-sofa::type::Vec6f& getVec(sofa::defaulttype::Vec6fTypes::Deriv& i){ return i; }
-const sofa::type::Vec6f& getVec(const sofa::defaulttype::Vec6fTypes::Deriv& i){ return i; }
-
 template<class DataTypes>
 bool FixedPlaneConstraint<DataTypes>::isPointInPlane(Coord p) const
 {
-    Vec<Coord::spatial_dimensions,Real> pos = getVec(p) ;
-    Real d=pos*getVec(d_direction.getValue());
-    if ((d>d_dmin.getValue())&& (d<d_dmax.getValue()))
+    Real d = getVec(p) * getVec(d_direction.getValue());
+    if ((d>d_dmin.getValue()) && (d<d_dmax.getValue()))
```

Comment:
```suggestion
    return d > d_dmin.getValue() && d < d_dmax.getValue();
```
and remove the `return`'s in the next lines

Reply:
👍